components/CollectionCells/src/MDCCollectionViewCell.m

```diff
@@ -58,6 +68,10 @@ @interface MDCAccessoryTypeImageView : UIImageView
 @implementation MDCAccessoryTypeImageView
 @end
 
+@interface MDCCollectionViewCell ()
+@property (nonatomic, assign) MDCCollectionViewCellSelectionState selectionState;
```

Comment:
I 100% agree with the motivation, but I'm concerned about adding state that mirrors existing state.

As an alternative, could we keep class / static references to the images in question?
static UIImage *s_checkCircleImage = [MDCIcons imageFor_ic_check_circle]; // Through dispatch_once
Then the flow would be:
if (selected) {
  if (_editingSelectorImageView.image != s_checkCircleImage) {
    _editingSelectorImageView.image = s_checkCircleImage;
    _editingSelectorImageView.tintColor = self.editingSelectorColor;
  } 
  self.accessibilityTraits |= UIAccessibilityTraitSelected;
}

This may not be a good idea, why don't we catch up in person tomorrow?